src/AudioGeneratorMP3.h

```diff
@@ -32,6 +32,7 @@ class AudioGeneratorMP3 : public AudioGenerator
     AudioGeneratorMP3(void *preallocateSpace, int preallocateSize);
     virtual ~AudioGeneratorMP3() override;
     virtual bool begin(AudioFileSource *source, AudioOutput *output) override;
+    virtual bool begin(AudioFileSource *source, AudioOutput *output, bool replay) override { replayOnEnd = replay; return begin(source, output); };
```

Comment:
Actually, since you've got this in the AudioGenerator base class, this definition isn't needed at all.  The default virt table one will work fine, no?

Reply:
I would have thought so too, but for some reason the compiler complains without this. Granted, I haven't done much C++ in a while, so I may be missing some nuances of how virtual functions and overriding in base classes work. It doesn't seem to like overloading the begin() method:

```
In file included from /home/ryan/Arduino/libraries/Bluesaber/src/Bluesaber.h:26:0,
                 from /home/ryan/Arduino/bluesaber/bluesaber.ino:10:
/home/ryan/Arduino/libraries/Bluesaber/src/SoundService.h: In member function 'void SoundService::play(const char*, bool)':
/home/ryan/Arduino/libraries/Bluesaber/src/SoundService.h:56:42: error: no matching function for call to 'AudioGeneratorMP3::begin(AudioFileSourceSPIFFS*&, AudioOutputI2S*&, bool&)'
       if (mp3->begin(newFile, out, repeat)) {
                                          ^
In file included from /home/ryan/Arduino/libraries/Bluesaber/src/SoundService.h:27:0,
                 from /home/ryan/Arduino/libraries/Bluesaber/src/Bluesaber.h:26,
                 from /home/ryan/Arduino/bluesaber/bluesaber.ino:10:
/home/ryan/Arduino/libraries/ESP8266Audio/src/AudioGeneratorMP3.h:34:18: note: candidate: virtual bool AudioGeneratorMP3::begin(AudioFileSource*, AudioOutput*)
     virtual bool begin(AudioFileSource *source, AudioOutput *output) override;
                  ^
/home/ryan/Arduino/libraries/ESP8266Audio/src/AudioGeneratorMP3.h:34:18: note:   candidate expects 2 arguments, 3 provided
```